### ISSUE TITLE
Materialize given queries in the graph

### ISSUE THREAD

**Issue by jnguyenx** [SciGraph/SciGraph#224]
@mbrush @benwbooth @kshefchek @cmungall 

The background https://github.com/monarch-initiative/monarch-cypher-queries/issues/7

The idea is to have a step after the graph load, that would take cypher queries from the user's configuration and materialize edges into the graph. The user would provide the cypher queries and the relationship type to materialize.

A fictional example:
```
# User configuration
hasPhenotype
(:variant)-[:foo]->(:gene)-[:bar]->(:phenotype)
```

After materializing the edges, the graph should have:
```
(:variant)-[:hasPhenotype]->(:phenotype)
```

**Comment by kshefchek** [comment]
While I see the benefit of having this done as a SciGraph post processor, I think it's easy enough to do this in cypher, for example:

    MATCH path=(subject:gene)<-[:foo*0..1]-(feature)-[:bar|baz*2..]->(object:Phenotype)
    WITH extract(edge IN relationships(path)| id(edge)) as relations, subject, object, path
    MERGE (subject)-[edge:`http://purl.obolibrary.org/obo/RO_0002200`]->(object)
    ON CREATE SET edge.qualifer = 'inferred', edge.path = relations, edge.lbl = 'has phenotype'
    ON MATCH  SET edge.path = edge.path + relations

This would merge multiple inferred edges into one, with an edge property that holds all the relationship ids that can recreate all the direct paths.  Then one could regenerate the direct paths either in the Java API or via 

    MATCH ()-[r]->()
    WHERE id(r) = 108134202
    WITH r.path as paths
    MATCH p=()-[edge]-()
    WHERE id(edge) IN paths
    RETURN p

Here it is in action:
![scigraph-example](https://user-images.githubusercontent.com/1467299/34901567-d30ddeda-f7c0-11e7-9fa8-195767ecbc78.gif)

**Comment by cmungall** [comment]
how did you make that video?

so it's not exactly a super-intuitive chunk of code, hence the idea to do it ahead of time, rather than having the person writing the cypher memorize or paste in the text each time.. unless I misunderstand?

**Comment by kshefchek** [comment]
I used a program called peek - https://github.com/phw/peek

My idea is that we write/maintain the cypher queries that create these inferred edges, and run them after the scigraph load and before indexing into solr.  Then we could simplify the solr queries and association graph munging since the edge ids will already be stored as edge properties for these inferred edges.

It's not super intuitive, but maybe more transparent than the java API route.  We could of course have some yaml configuration that is converted to cypher.

**Comment by cmungall** [comment]
ah ok, I guess I'd call that post-processing, I think we're on the same page

**Comment by kshefchek** [comment]
Yes, I think even adding functionality to configure a post processing step that executes cypher from a file (or directory) could be useful.
